modules/SofaSparseSolver/src/SofaSparseSolver/SparseCholeskySolver.cpp

```diff
@@ -108,10 +108,71 @@ void SparseCholeskySolver<TMatrix,TVector>::invert(Matrix& M)
     A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &A );
     tmp.resize(A.n);
-    S = cs_schol (&A, order) ;		/* ordering and symbolic analysis */
+    
+    perm.resize(A.n);
+    iperm.resize(A.n);
+
+    if(computePermutation)
+    {
+        fill_reducing_perm(A , perm.data(), iperm.data() ); // compute the fill reducing permutation
+        computePermutation = false;
+    }
+    S = symbolic_Chol( &A ); // symbolic analysis
+    permuted_A = cs_permute( &(A), iperm.data() , perm.data() , 1);
+
+    sofa::helper::ScopedAdvancedTimer factorizationTimer("factorization");
     N = cs_chol (&A, S) ;		/* numeric Cholesky factorization */
 }
 
+template<class TMatrix, class TVector>
+void SparseCholeskySolver<TMatrix,TVector>::fill_reducing_perm(cs A,int * perm,int * invperm)
```

Comment:
```suggestion
void SparseCholeskySolver<TMatrix,TVector>::fill_reducing_perm(const cs& A,int * perm,int * invperm)
```
pass by reference to avoid a copy